dtfs-central-api/src/services/utilisation-report-data-validator/index.ts

```diff
@@ -99,38 +99,34 @@ export const validateUtilisationReportCsvCellData = (
     generatePaymentExchangeRateError,
   ];
 
-  return csvData.flatMap((csvRow) => {
-    const csvDataErrors: UtilisationReportDataValidationError[] = [];
-
-    cellValidations.forEach(({ header, errorGenerator }) => {
-      if (availableHeaders.includes(header)) {
-        const error = errorGenerator(csvRow[header], csvRow.exporter?.value);
-        if (error) {
-          csvDataErrors.push(error);
+  const allErrors = await Promise.all(
+    csvData.map(async (csvRow) => {
+      const cellErrorPromises = cellValidations.map(({ header, errorGenerator }) => {
+        if (!availableHeaders.includes(header)) {
+          return null;
         }
-      }
-    });
 
-    rowValidations.forEach((errorGenerator) => {
-      const error = errorGenerator(csvRow);
-      if (error) {
-        csvDataErrors.push(error);
-      }
-    });
+        return errorGenerator(csvRow[header], csvRow.exporter?.value);
+      });
 
-    return csvDataErrors;
-  });
+      const rowErrorPromises = rowValidations.map((errorGenerator) => errorGenerator(csvRow));
+
+      return Promise.all([...cellErrorPromises, ...rowErrorPromises]);
+    }),
+  );
+
+  return allErrors.flat().filter((error): error is UtilisationReportDataValidationError => error !== null);
```

Comment:
can this be a const and return the const then

Reply:
Sure

Reply:
Have updated

Reply:
Resolving this - moved to a new comment (so we can clear the historic unresolved comments) https://github.com/UK-Export-Finance/dtfs2/pull/3462/files#r1750553315